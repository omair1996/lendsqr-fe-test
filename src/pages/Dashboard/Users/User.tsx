import UserDashboard from '@/components/userDashboard/UserDashboard';

const User = () => {
  return (
    <div>
      <UserDashboard />
    </div>
  );
};
export default User;
