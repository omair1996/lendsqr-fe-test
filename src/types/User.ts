export interface User {
  id: string;
  organization: string;
  username: string;
  email: string;
  phone: string;
  date_joined: string;
  status: string;
  profile: {
    avatar: string;
  };
  hasLoan: boolean;
  hasSavings: boolean;
}
