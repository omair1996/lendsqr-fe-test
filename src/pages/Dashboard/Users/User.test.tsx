import { render, screen } from '@testing-library/react';
import { describe, test, expect, vi } from 'vitest';
import User from './User';

// Mock the UserDashboard component
vi.mock('@/components/userDashboard/UserDashboard', () => ({
  default: () => <div data-testid="user-dashboard">User Dashboard</div>,
}));

describe('User Page', () => {
  test('renders without crashing', () => {
    render(<User />);
    expect(screen.getByTestId('user-dashboard')).toBeInTheDocument();
  });

  test('displays the UserDashboard component', () => {
    render(<User />);
    expect(screen.getByTestId('user-dashboard')).toHaveTextContent('User Dashboard');
  });

  test('matches snapshot', () => {
    const { asFragment } = render(<User />);
    expect(asFragment()).toMatchSnapshot();
  });
});
