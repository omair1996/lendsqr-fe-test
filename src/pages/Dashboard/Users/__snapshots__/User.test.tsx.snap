// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`User Page > matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      data-testid="user-dashboard"
    >
      User Dashboard
    </div>
  </div>
</DocumentFragment>
`;
